src/query/src/types.ts

```diff
@@ -132,6 +132,13 @@ export type SquatInsights = {
 
 export type ParserFn = (opt: ParserState) => Promise<ParserState>
 
+export type ParsedSelectorToken = {
+  token: string
+  type: 'pseudo' | 'attribute' | 'class' | 'combinator' | 'id'
+  key?: string
+  value?: string
+}
+
```

Comment:
Extending the ast node here will preserve info that I think is going to make your life easier traversing the nodes later in the query-token component.

```suggestion
export type ParsedSelectorToken = PostcssNode & {
  token: string
}

```